spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/ConfigurationPropertiesBinder.java

```diff
@@ -252,8 +252,31 @@ ConfigurationPropertiesBinder create() {
 					? target.withBindRestrictions(BindRestriction.NO_DIRECT_PROPERTY) : target;
 		}
 
-		private boolean isConfigurationProperties(Class<?> target) {
-			return target != null && MergedAnnotations.from(target).isPresent(ConfigurationProperties.class);
+		private boolean isConfigurationProperties(final Class<?> target) {
+			return target != null && (hasConfigurationProperty(target) || isNestedClassOfConfigurationProperty(target));
+		}
+
+		private boolean hasConfigurationProperty(final Class<?> target) {
```

Comment:
Did you intentionally express the `ConfigurationProperty` in the singular form? Since `ConfigurationProperties` is the correct annotation name, isn't it better to use this term consistently?